### PR TITLE
Kc 5816 scala dataset error

### PR DESCRIPTION
**Pull request by etmarx**
Fixes problems with scala dataset word count transform

### CODE REVIEW

File: analytics/scala/src/main/scala/com/koverse/example/spark/WordCountDatasetTransform.scala [hunk 1]
```diff
@@ -16,19 +16,51 @@
...
+  }
+
+  def getId: Long = id
+
+  def setId(id: Long): Unit = {
+    this.id = id
+  }
+}
+
+case class WordCount(var text: String, var count: Int) extends Serializable {
```
Comment (bmcrea) [inline analytics/scala/src/main/scala/com/koverse/example/spark/WordCountDatasetTransform.scala]: You don't need 'var' here. Just text:String etc.

File: analytics/scala/src/main/scala/com/koverse/example/spark/WordCountDatasetTransform.scala [hunk 2]
```diff
@@ -16,19 +16,51 @@
...
+
+  def this() = this("", 0)
+
+  def getText: String = text
```
Comment (bmcrea) [inline analytics/scala/src/main/scala/com/koverse/example/spark/WordCountDatasetTransform.scala]: Getters and setters are added by the compiler in scala and usually not specified. Did you find you needed them?

File: analytics/scala/src/main/scala/com/koverse/example/spark/WordCounter.scala [hunk 3]
```diff
@@ -61,7 +61,10 @@ class WordCounter(
     import spark.implicits._
 
     // for each Record, tokenize the specified text field and count each occurrence
-    val wordCountDataset = inputRecordsDataset.flatMap(record => record.text.toLowerCase.split(tokenizationString))
+    inputRecordsDataset.show()
+    // Take the column that contains the text and tokenize and count the words
+//    val wordDF = inputDataFrame.explode(textFieldName, "word") { text: String => text.split(tokenizationString) }
```
Comment (bmcrea) [inline analytics/scala/src/main/scala/com/koverse/example/spark/WordCounter.scala]: If this isn't needed, please remove it.

File: analytics/scala/src/test/resources/client.properties [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # Please register your application with your copy of Koverse before running
 # the examples in this package. To register your application follow the instructions at TODO
-client.name=examples
-client.secret=changeme
+client.name=testClient
+client.secret=c4243d17-e320-470d-b534-f9ea238e8adb
```
Comment (bmcrea) [inline analytics/scala/src/test/resources/client.properties]: I dont' think we should use the test client here. That's our secret. This repo is for customers. They should add their own client name and secret here

File: analytics/scala/src/main/scala/com/koverse/example/spark/WordCountDatasetTransform.scala [hunk 5]
```diff
@@ -16,19 +16,51 @@
...
 package com.koverse.example.spark
 
+import java.io.Serializable
+
 import com.koverse.sdk.Version
 import com.koverse.sdk.data.Parameter
 import com.koverse.sdk.transform.scala.{DatasetTransform, DatasetTransformContext}
 import org.apache.spark.sql.Dataset
 
+case class Message(var article: String, var id: Long) extends Serializable {
```
Comment (bmcrea) [inline analytics/scala/src/main/scala/com/koverse/example/spark/WordCountDatasetTransform.scala]: same comments as WordCount case class

File: analytics/scala/src/main/scala/com/koverse/example/spark/WordCountDatasetTransform.scala [hunk 6]
```diff
@@ -16,15 +16,57 @@
...
 package com.koverse.example.spark
 
+import java.io.Serializable
+
 import com.koverse.sdk.Version
 import com.koverse.sdk.data.Parameter
 import com.koverse.sdk.transform.scala.{DatasetTransform, DatasetTransformContext}
 import org.apache.spark.sql.Dataset
 
+case class Message(var article: String, var title: String, var id: Long) extends Serializable {
```
Comment (bmcrea) [inline analytics/scala/src/main/scala/com/koverse/example/spark/WordCountDatasetTransform.scala]: Scala case classes extend Serializable by default. You shouldn't need that there.

File: analytics/scala/src/main/scala/com/koverse/example/spark/WordCountDatasetTransform.scala (hunk 2 above)
Comment (etmarx) [reply analytics/scala/src/main/scala/com/koverse/example/spark/WordCountDatasetTransform.scala]: Yes, I found that I needed them for getting the writeMethods in KoversePOJOConverter. I tried all kinds of alternatives and this was the only thing that worked.

File: analytics/scala/src/test/resources/client.properties (hunk 4 above)
Comment (etmarx) [reply analytics/scala/src/test/resources/client.properties]: absolutely. I forgot to change it back.

File: analytics/scala/src/main/scala/com/koverse/example/spark/WordCountDatasetTransform.scala (hunk 6 above)
Comment (etmarx) [reply analytics/scala/src/main/scala/com/koverse/example/spark/WordCountDatasetTransform.scala]: Ok, I will remove

File: analytics/scala/src/main/scala/com/koverse/example/spark/WordCounter.scala (hunk 3 above)
Comment (etmarx) [reply analytics/scala/src/main/scala/com/koverse/example/spark/WordCounter.scala]: I don't see it there. Maybe you were looking before my latest push?

File: analytics/scala/src/main/scala/com/koverse/example/spark/WordCountDatasetTransform.scala (hunk 1 above)
Comment (etmarx) [reply analytics/scala/src/main/scala/com/koverse/example/spark/WordCountDatasetTransform.scala]: Actually, I do need it since I added getters/setters

**Comment by etmarx** [comment]
Merged changes into KC-5779 since that ticket is still being worked

**Pull request closed without merging**